Code/GraphMol/SubstructLibrary/SubstructLibrary.cpp

```diff
@@ -86,6 +89,64 @@ struct Bits {
     }
   }
 
+  // FIX complete this
+  Bits(const FPHolderBase *fingerprints, const ExtendedQueryMol &xqm,
+       const SubstructMatchParameters &ssparams)
+      : fps(fingerprints), params(ssparams) {
+    if (fps) {
+      const auto *tph = dynamic_cast<const TautomerPatternHolder *>(fps);
+      const auto *ph = dynamic_cast<const PatternHolder *>(fps);
+      if (std::holds_alternative<ExtendedQueryMol::RWMol_T>(xqm.xqmol)) {
+        queryBits = fps->makeFingerprint(
+            *std::get<ExtendedQueryMol::RWMol_T>(xqm.xqmol));
+      } else if (std::holds_alternative<ExtendedQueryMol::MolBundle_T>(
+                     xqm.xqmol)) {
+        auto &bndl = std::get<ExtendedQueryMol::MolBundle_T>(xqm.xqmol);
+        auto tqb = new ExplicitBitVect(ph->getNumBits());
+        queryBits = tqb;
+        for (auto mol : bndl->getMols()) {
+          auto tfp = fps->makeFingerprint(*mol);
+          *tqb &= *tfp;
+          delete tfp;
+        }
+      } else if (std::holds_alternative<ExtendedQueryMol::TautomerQuery_T>(
+                     xqm.xqmol)) {
+        auto &tq = std::get<ExtendedQueryMol::TautomerQuery_T>(xqm.xqmol);
+        if (!tph) {
+          BOOST_LOG(rdWarningLog) << "Pattern fingerprints for tautomersearch "
+                                     "aren't tautomer fingerprints, ignoring..."
+                                  << std::endl;
+          queryBits = nullptr;
+          fps = nullptr;
+        } else {
+          queryBits = tq->patternFingerprintTemplate(tph->getNumBits());
+        }
+      } else if (std::holds_alternative<ExtendedQueryMol::TautomerBundle_T>(
+                     xqm.xqmol)) {
+        if (!tph) {
+          BOOST_LOG(rdWarningLog) << "Pattern fingerprints for tautomersearch "
+                                     "aren't tautomer fingerprints, ignoring..."
+                                  << std::endl;
+          queryBits = nullptr;
+          fps = nullptr;
+        } else {
+          auto &bndl = std::get<ExtendedQueryMol::TautomerBundle_T>(xqm.xqmol);
+          auto tqb = new ExplicitBitVect(ph->getNumBits());
```

Comment:
It doesn't look like tqb is being used or freed.

Reply:
Ah, this is really queryBits.

Reply:
yeah, and we need the temporary variable since `queryBits` is const